src/view/domconverter.js

```diff
@@ -243,22 +247,41 @@ export default class DomConverter {
 	 * @returns {Iterable.<Node>} DOM nodes.
 	 */
 	* viewChildrenToDom( viewElement, domDocument, options = {} ) {
-		const fillerPositionOffset = viewElement.getFillerOffset && viewElement.getFillerOffset();
+		const blockFillerOffset = viewElement.getFillerOffset && viewElement.getFillerOffset();
+		const inlineFillerOffset = options.inlineFiller && options.inlineFiller.parent == viewElement ? options.inlineFiller.offset : null;
 		let offset = 0;
 
 		for ( const childView of viewElement.getChildren() ) {
-			if ( fillerPositionOffset === offset ) {
+			if ( blockFillerOffset === offset ) {
 				yield this.blockFiller( domDocument );
 			}
 
-			yield this.viewToDom( childView, domDocument, options );
+			const domNode = this.viewToDom( childView, domDocument, options );
+
+			// We need to insert inline filler before the dom node.
+			if ( inlineFillerOffset === offset ) {
```

Comment:
AFAI understand, this is incorrect scenario. Filler can be only at the beginning of the text node. Filler should not be needed in this case, and if it was already in the DOM, it will be just preversed (at the beginning of the text node).